plutus-benchmark/flat/Main.hs

```diff
@@ -0,0 +1,86 @@
+{-# LANGUAGE BangPatterns      #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main where
+
+import           Control.DeepSeq              (force)
+import           Criterion.Main
+import           Criterion.Types
+import qualified Data.ByteString              as BS
+import           Data.Map.Strict              (Map, empty, insert, lookup)
+import           Data.Maybe                   (fromJust)
+import           Data.Text                    (Text, unpack)
+import           System.Mem                   (performMajorGC)
+
+import           Language.PlutusCore          (Name (..))
+import           Language.PlutusCore.Universe
+import           Language.UntypedPlutusCore
+
+import           Codec
+import           Dataset
+import           Report
+
+import           Prelude                      hiding (lookup)
+
+type Tm = Term Name DefaultUni ()
+
+serializedContracts :: Map (Text, Text) BS.ByteString
+serializedContracts =
+  foldl update empty [ (contract, codec) | contract <- contracts, codec <- codecs ]
+  where
+    update :: Map (Text, Text) BS.ByteString
+           -> ((Text, Tm), (Text, Codec Tm))
+           -> Map (Text, Text) BS.ByteString
+    update cs ((contractName, contract), (codecName, codec)) =
+      insert (contractName, codecName) ((Codec.serialize codec) contract) cs
+
+benchmarkSerialize ::
+     (Text, Tm)
+  -> (Text, Codec Tm)
+  -> Benchmark
+benchmarkSerialize (contractName, contract) (codecName, (Codec serialize' _)) =
+  bench (unpack $ " " <> contractName <> " - " <> codecName)
+        -- (nf (BS.length . serialize) contract) ~ original code
```

Comment:
I dislike leaving dead code in. I'd either delete it, or make a real comment out of it.

Reply:
I added a comment asking if anyone sees a problem with me removing some of the code from the original benchmarks. In this case, it's the call to the `length` function.